explorer/interpreter/impl_scope.cpp

```diff
@@ -137,6 +141,41 @@ auto ImplScope::ResolveInterface(Nonnull<const InterfaceType*> iface_type,
   return *result;
 }
 
+// Combines the results of two impl lookups.
+static auto CombineResults(Nonnull<const InterfaceType*> iface_type,
+                           Nonnull<const Value*> type,
+                           SourceLocation source_loc,
+                           std::optional<Nonnull<const Witness*>> a,
+                           std::optional<Nonnull<const Witness*>> b)
+    -> ErrorOr<std::optional<Nonnull<const Witness*>>> {
+  // If only one lookup succeeded, return that.
+  if (!a) {
+    return b;
+  }
+  if (!b) {
+    return a;
+  }
+  // If either of them was a symbolic result, then they'll end up being
+  // equivalent. Pick whichever we found first.
+  if (!isa<ImplWitness>(*b)) {
+    return a;
+  }
```

Comment:
In the above test, it didn't matter which order we did the comparsion in, because you get the same result either way. Swapped the above test anyway to keep the code more consistent.

On reflection, I don't think we really want first/second semantics here, just an arbitrary tie-breaker, so I've updated the comments to say so and kept the `a` / `b` naming.